### DISCUSSION TITLE
Spring restraint in one axis only

### DISCUSSION THREAD

**Discussion by daveunwin** [DualSPHysics/DualSPHysics#198] [DUALSPHYSICS_V5.2]
Hi all,
I'm sure there is a relatively straight forward way to do this but I'm trying to implement a spring restraint on a floating body using chrono that only acts as a spring in a fixed direction. All the examples I have seen show the spring varying in both the X and Z directions (for a 2D analysis). In my case, the floating body needs to move without restraint vertically but to a defined stiffness horizontally.  I have tried defining a second 'dummy' body that is restricted to movement only in the vertical direction but still seem to be having problems achieving want I am after. Ideally it would be possible to specifiy a stiffness when restricting translation in the point_line constraint (it appears possible to specify a rotational stiffness there but not linear?).
Can anyone help with a way around?
Cheers

**Comment by DualSPHysics** [comment]
I think you should try to combine your spring with the restriction in Chrono named link_pointline.

Documentation: page 10 of XML_GUIDE_CHRONO.pdf in DualSPHysics_v5.2\doc\guides

Examples in:
- examples\chrono\06_ZipLine
- examples\chrono\10_PointAbsorber
- 
Regards
